Net/src/SocketImpl.cpp

```diff
@@ -1372,12 +1372,12 @@ void SocketImpl::error(int code, const std::string& arg)
 		throw IOException(NumberFormatter::format(code), arg, code);
 	}
 }
-
+#ifdef POCO_HAVE_SENDFILE
```

Comment:
Still why do we not get rid of whole 

`throw Poco::NotImplementedException("sendfile not implemented for this platform");`

Then we just use read and normal socket to do send if that is not supported. Then user really does not have to think about these things. They just use our sendFile and it will work everytime. This is probably what they want as they use Poco which is multiplatform library.

Reply:
This behavior was approved with @aleks-f https://github.com/pocoproject/poco/pull/4007#discussion_r1419524268

Reply:
Hmm. Ok. Seems strange to me as now users also need to do ifdef rules to use Poco. But if that was chosen already we go with that.